### PR TITLE
Convert version resolver specs to projects

### PR DESCRIPTION
**Pull request by jurre**
This was mostly an automated pass with some manual cleanup.

The project names are slightly gnarly, and I think we'll want to name
them ourselves eventually, but I figured that keeping them as is until
we've converted all specs will make it easier to do a single rename pass
once we're done.

### CODE REVIEW

File: bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb [hunk 1]
```diff
@@ -47,68 +46,42 @@
...
   describe "#latest_resolvable_version_details" do
     subject { resolver.latest_resolvable_version_details }
 
     context "with a rubygems source" do
       context "with a ~> version specified constraining the update" do
-        let(:gemfile_fixture_name) { "Gemfile" }
-        let(:lockfile_fixture_name) { "Gemfile.lock" }
         let(:requirement_string) { "~> 1.4.0" }
 
+        let(:dependency_files) { project_dependency_files("bundler1/Gemfile_Gemfile") }
```
Comment (feelepxyz) [inline bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb:58]: Would it work if we uniqued these names so this ends up as `Gemfile`?

**Review by feelepxyz — COMMENTED** [review]
Nice one getting it working for bundler! Added a comment about uniquing manifest and lockfile so we don't end up with duplicates as this seems like the most common pattern, otherwise 🙌 💯

**Comment by feelepxyz** [comment]
> The project names are slightly gnarly, and I think we'll want to name
> them ourselves eventually, but I figured that keeping them as is until
> we've converted all specs will make it easier to do a single rename pass
> once we're done.

Seems ok to pass through and re-name after but wondering if a `uniq` would work in most cases?

File: bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb (hunk 1 above)
Comment (jurre) [reply bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb:58]: Yeah, but we need to also handle the scenario where there's only a Gemfile too

**Comment by brrygrdn** [comment]
+1 to keeping the renaming until after, as long as we're using a consistent project naming strategy that seems fine, we _could_ add the extension as an underscore to make them a little easier to read, but ultimately I don't think it has much value.

**Review by feelepxyz — APPROVED** [review]
👍 :shipit:

File: bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb (hunk 1 above)
Comment (jurre) [reply bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb:58]: Maybe we can figure out if there's a lockfile, and if not we append `_no_lockfile`

**Comment by jurre** [comment]
@brrygrdn @feelepxyz I've improved the fixture names, thanks for calling it out, this is much nicer and it wasn't much effort

**Review by brrygrdn — APPROVED** [review]

**Pull request merged**